src/ResultWriter/ReceiverWriter.cpp

```diff
@@ -47,6 +47,44 @@
 #include <Parallel/MPI.h>
 #include <Modules/Modules.h>
 
+#include <sstream>
+#include <string>
+#include <fstream>
+#include <regex>
+
+Eigen::Vector3d seissol::writer::parseReceiverLine(const std::string& line) {
+  std::regex rgx("\\s+");
+  std::sregex_token_iterator iter(line.begin(),
+                                  line.end(),
+                                  rgx,
+                                  -1);
+  std::sregex_token_iterator end;
+  std::array<double, 3> coordinates{};
+  unsigned numberOfCoordinates = 0;
+  for (; iter != end; ++iter, ++numberOfCoordinates) {
+    if (numberOfCoordinates > coordinates.size()) {
+      break;
+    }
+    coordinates[numberOfCoordinates] = std::stod(*iter);
+  }
+  if (numberOfCoordinates != coordinates.size()) {
+    throw std::runtime_error("Incorrect number of coordinates in line " + line + ".");
+  }
+  return {coordinates[0], coordinates[1], coordinates[2]};
```

Comment:
Why do you have the `std::array` as intermediate representation and not an `Eigen::Vector3d` from the beginning?

Reply:
A very good question indeed...